spark/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -652,9 +652,12 @@ class DeltaAnalysis(session: SparkSession)
             Identifier.of(Array("delta"), path.toString))
         }
         // Trying to clone something on itself should be a no-op
-        if (sourceTbl == new CloneDeltaSource(targetTbl)) {
-          return LocalRelation()
+        sourceTbl match {
+          case deltaSourceTbl: CloneDeltaSource if deltaSourceTbl.sourceTable == targetTbl =>
```

Comment:
shouldnt this be `cloneSource` instead of `deltaSourceTbl` as then `cloneSource.sourceTable` will make more sense than effectively using `someTable.anotherTable`?

Reply:
You mean, something like this?
```suggestion
          case cloneSource: CloneDeltaSource if cloneSource.sourceTable == targetTbl =>
```

Reply:
yeah.